src/main/java/org/jenkinsci/plugins/credentialsbinding/impl/CertificateMultiBinding.java

```diff
@@ -0,0 +1,131 @@
+package org.jenkinsci.plugins.credentialsbinding.impl;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+
+import org.jenkinsci.plugins.credentialsbinding.BindingDescriptor;
+import org.jenkinsci.plugins.credentialsbinding.MultiBinding;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import com.cloudbees.plugins.credentials.common.StandardCertificateCredentials;
+
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+
+public class CertificateMultiBinding extends MultiBinding<StandardCertificateCredentials> {
+
+	private final String keystoreVariable;
+
+	public String getKeystoreVariable() {
+		return keystoreVariable;
+	}
+
+	public String getPasswordVariable() {
+		return passwordVariable;
+	}
+
+	public String getAliasVariable() {
+		return aliasVariable;
+	}
+
+	private final String passwordVariable;
+	private final String aliasVariable;
+
+	@DataBoundConstructor
+	public CertificateMultiBinding(String keystoreVariable, String passwordVariable, String aliasVariable,
+			String credentialsId) {

```

Comment:
`keystoreVariable` is likely mandatory, but `passwordVariable` and `aliasVariable` seem optional, so it would be better to specify these with `@DataBoundSetter`. Would allow them to be omitted in a Pipeline job using `withCredentials`.